src/dhcp/server.c

```diff
@@ -675,7 +675,7 @@ static gboolean listener_event(GIOChannel *channel, GIOCondition condition,
 
 	server_id_option = dhcp_get_option(&packet, DHCP_SERVER_ID);
 	if (server_id_option) {
-		uint32_t server_nid = get_be32(server_id_option);
+		uint32_t server_nid = get_unaligned((uint32_t *) server_id_option);
```

Comment:
Let me be explicit. That's what it does, but... what it fix? (since linux/android connection is working, what is the use case involved on this fix) That's what must be in the commit to allow a rebase instead an non fast forward merge.